tests/integration/helpers/cluster.py

```diff
@@ -128,11 +128,20 @@ def run_and_check(
     return out
 
 
+# get_free_port() can return the same port multiple times.
+# To avoid it we just collect all returned ports and check for overlap.
+used_ports = []
```

Comment:
Unfortunately in CI run tests in parallel within the same batch. For example for `Integration tests (release) [4/6]` we will run several `pytest` processes inside the docker container (yes, we several layers of parallelism). So, the only good solution would be just to synchronize this process via disk. I was thinking about using `sqlite`... and just specify a path for each this standalone `pytest`.

Reply:
Or we can revive https://github.com/ClickHouse/ClickHouse/pull/55055 and rebind the port.

Reply:
Why do we need this several levels of parallelism? Maybe run separate Docker containers for each test?

Reply:
Okey, probably I found case when there was a clash between parallel runs: https://s3.amazonaws.com/clickhouse-test-reports/0/aabf7a576ce42ef03456bf846784adfd1635ccaa/integration_tests__release__[3_4]//home/ubuntu/actions-runner/_work/_temp/test/output_dir/integration_run_parallel5_0.log.
Let's revive rebinding indeed, will add in this PR

Reply:
I've assigned port rebinding to the @fm4v already...

Reply:
No problem, then I will pick something else. We had a clash on fixing ports clashing :)